docs/src/main/asciidoc/spring-cloud-gateway.adoc

```diff
@@ -874,6 +874,13 @@ spring:
 ----
 
 In either case, the HTTP status of the response will be set to 401.
+Example configuration to produce http header:
+----
+# by default originalStatusHeaderName = original-http-status
+spring.cloud.gateway.setStatus.originalStatusHeaderName=my-backward-compatible-name
+# by default addOriginalStatus has false value
+spring.cloud.gateway.setStatus.addOriginalStatus=true
```

Comment:
This could be entirely eliminated and if the default header name was null and just checked for the existence of the header name. @ryanjbaxter what do you think?

Reply:
@spencergibb  So if no header name is configured we wouldnt add anything to the response?

Reply:
Correct, rather than a boolean and a header name, just header name

Reply:
Sounds good to me

Reply:
I made it this way to have "default header name" ability.
But if you think that it should be omitted I can remove this boolean flag and use absent value as a flag.

Reply:
Since we're making this opt in, it will be very clear what the header name will be, so yes please do so.

Reply:
done